### PR TITLE
Parole

### PR DESCRIPTION
**Pull request by Kyep**
Goals:
- Encourage security to grant parole to prisoners, enabling them to interact with other players and contribute to the station, rather than just wasting their lives away in the brig.

Changes:
- Tracking implants now show the implanted person's name, general location (e.g: "Medbay Maintenance") and health status on prisoner monitoring computers. This makes them far more useful.
- ~~Parole implants have been added. These are chem implants pre-filled with non-addictive sedative (ether). A violent person wants parole? Give him one of these, plus a tracking implant, and you'll be able to sedate and retrieve him later if he turns violent again. This is technically already possible with general-purpose chem implants, but (A) nobody ever uses them, as they're too much hassle, and (B) prisoners fear being given lethal chem implants. This PR addresses both of those issues. Once activated from a prisoner management console, these parole implants can create blurry vision (after 30s), then unconsciousness (after a minute). At maximum dosage, the unconsciousness lasts approximately 3-4 minutes. These implants have a set amount of chems they can use - once those are depleted, the implant is absorbed by the body.~~ Parole implants have been deleted from this PR.
- ~~Parole jumpsuits have been added. These are yellow jumpsuits which, when worn with an ID and locked with a jumpsuit control tool (also added), neither they nor the ID the person is wearing can be removed until unlocked again with the tool. They prevent parolees disguising their identity (must always wear their ID), or turning off suit sensors. Note: traitors can still escape these easily with an emag. Even normal crew can potentially escape if they're lucky enough to find a reliable source of EMPs (e.g: engine, EXPERIMENTOR). Breaking the parole suit with emag or EMP will be noticed, though.~~ Parole jumpsuits have been deleted from this PR.
- ~~Added a parole box to the warden's locker. It contains two parole implants, two tracking implants, two parole jumpsuits, and one parole jumpsuit remote.~~

Notes:
- This PR does not seek to change SOP/Space Law. Conditions under which parole can be granted remain the same.
- This PR just makes parole a more useful/attractive option, thus hopefully resulting in more interactivity amongst players and less people bored out of their minds in the brig.

:cl: Kyep
tweak: Tracking implants are now useful, showing the implanted person's general location, and health status, on prisoner consoles.
fix: The NODROP flag now works correctly on jumpsuits.
/:cl:

### CODE REVIEW

**Comment by CrAzYPiLoT-SS13** [comment]
Yes, please. :+1:

**Comment by sasanek12** [comment]
100 times yes :+1:

**Comment by KasparoVy** [comment]
Wow, that's a pretty cool idea. Great work :+1:

**Comment by M3hillus** [comment]
:+1: I like it. A bit iffy on the locking jumpsuit, might be just me. Maybe have it alarmed?
 Or, instead of an entire implant, some sort of house arrest ankle bracelet.

**Comment by IcyV** [comment]
What are your thoughts on Security potentially using this to just full power game and put one of these suits on someone that remotely acts up? I personally don't see this as solving an issue of giving prisoners more interaction as much as I predict it will be used to just keep any potential problems instantly subdued.

I do like M3hi's idea of an ankle bracelet though. Maybe instead of a full suit it can be along those lines and alert security should a prisoner enter certain areas or tamper with it.

**Comment by FalseIncarnate** [comment]
Ankle bracelet could work, but then has issues with how it is put on them as we don't have a slot for "accessories" aside from the one on jumpsuits. But all they'd need to do is take off the suit to lose the accessory, which means you'd be better off merging it into a special parolee suit that they can't remove alone. Wouldn't want them getting in trouble for changing into their uniform because they got a job after all.

File: code/game/machinery/computer/prisoner.dm [hunk 1]
```diff
@@ -58,12 +57,17 @@
 			Tr = get_turf(T)
 			if((Tr) && (Tr.z != src.z))	continue//Out of range
 			if(!T.implanted) continue
-			var/loc_display = "Unknown"
 			var/mob/living/carbon/M = T.imp_in
+			var/loc_display = "Unknown"
+			var/health_display = "OK"
+			var/total_loss = round(M.getOxyLoss() + M.getToxLoss() + M.getFireLoss() + M.getBruteLoss(), 1)
+			if (M.stat > UNCONSCIOUS)
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/prisoner.dm]: `if(M.stat == DEAD)`

File: code/modules/clothing/under/color.dm [hunk 2]
```diff
@@ -103,6 +103,55 @@
 	item_color = "yellow"
 	flags = ONESIZEFITSALL
 
+
+/obj/item/clothing/under/color/yellow/parole
+	name = "parole jumpsuit"
+	desc = ""
```
Comment (Fox-McCloud) [inline code/modules/clothing/under/color.dm]: Better to just have the unlocked description by default, then have the description change when it's impacted by various conditions.

and use `initial(desc)` when resetting it back.

**Comment by theColdflame** [comment]
I can see this being used by security to protect their own IDs, are there any measures which make that less attractive?

**Comment by Kyep** [comment]
@M3hillus
The lock can be defeated with emag, any repeated source of EMP, etc.
Jumpsuit was chosen rather than ankle bracelet as jumpsuit exists, jumpsuits already support sensors, and jumpsuits are visible.

@IcyV
I did consider the idea of generating an alert if the parolee entered certain areas.
Decided not to do it in this PR. Might put it in a different PR in future.

@IcyV
There are only two, its impossible to obtain more, sec has more than two troublemakers per shift, and the worst troublemakers (antags) are the ones most likely to break the suits. For all of these reasons sec cannot use them on everyone.

@Fox-McCloud
Changed

@theColdflame
Parole jumpsuits will no longer lock if the parolee is wearing an ID with sec or head access.

**Comment by M3hillus** [comment]
Perhaps add cutting them off, where it acts like handcuffs when wirecutters are used on the suit.

Using the same timer as handcuffs, and lets out an alarm/location of the user on the security channel a la beepsky when he arrests someone.

**Comment by Kyep** [comment]
@M3hillus Enabling anyone to get them off with wirecutters would make them pointless. So, not doing that.

**Comment by FlattestGuitar** [comment]
But it makes sense. Add an alert on the security radio "X has disabled their parole equipment in Y" and it Will give people an option to escape it without being a traitor or having access to SciChem.

**Comment by Kyep** [comment]
Security will find out they've disabled it anyway, when they go back to sec and sec tries to unlock it - finding it already unlocked.

Enabling them to break it with wirecutters makes it too easy, and a radio alert makes it too strong, because that way sec will be instantly on their ass if they break it, rather than them having a head start.

**Comment by Krausus** [comment]
> Conditions under which parole can be granted remain the same.

Where are the conditions for this actually specified?

> There are only two, its impossible to obtain more, sec has more than two troublemakers per shift, and the worst troublemakers (antags) are the ones most likely to break the suits. Any parole suits broken cannot be replaced. For all of these reasons sec simply cannot use them on everyone.

So, these are supposed to encourage security to grant parole to more prisoners, but they're so powerful that you have to limit security to two of them. And any actual traitors are likely to escape from them anyway, so why bother letting them walk out of security at all?

> neither they nor the ID the person is wearing can be removed

So the person is screwed out of changing jobs, using guest pass computers, wearing guest passes, mining... probably other stuff, since IDs were never intended to be stuck to you. And it's not as if the warden will go out of his way to help someone take off their ID, since apparently...

> This is technically already possible with general-purpose chem implants, but nobody ever uses them, as they're too much hassle

... security is too lazy to walk on down to the chemist and ask for some ether. Which you're compensating for by giving them a couple free 3-minute-knockout implants.

Considering we already have tracking implants to show rough location, and chem implants to knock people out, this just feels like "security hard, plz buff".

**Comment by DaveTheHeadcrab** [comment]
I do not like the idea of nodrop jumpsuits as sec general-issue.
At all.

**Comment by Kyep** [comment]
@Krausus 
Re: Conditions, see SOP. 
Re: Frequency of parole, currently parole happens VERY rarely. If these have any effect at all it will be an improvement.
Re: Changing jobs, better to be unable to change jobs for 10m than to be stuck in a brig cell for that time. Parolees can still use guest passes. Parolees can still mine (doesn't stop you changing hardsuit). 
Re: Security difficulty, this PR isn't about buffing security. In any situation these would be used, Security already have the right to lock someone in a tiny room for X minutes for their actions. This PR is about giving them incentive to try paroling the person instead. So that person can have more fun in the round, rather than wasting time in a very small prison cell. Security *give up* some control for this - there is more potential for the person to escape. Yet, it makes for a more fun round for everyone. I consider that very much worth it.

File: code/modules/clothing/under/color.dm [hunk 3]
```diff
@@ -103,6 +103,56 @@
...
+	sensor_mode = 3
+	var/parole_locked = 0
+	var/emagged = 0
+	var emped = 0
+	var/obj/item/weapon/card/id/parole_id = null
+	var/mob/living/carbon/human/worn_by = null
+
+/obj/item/clothing/under/color/yellow/parole/emag_act(user as mob)
+	if(parole_locked && !emagged)
+		to_chat(user, "\red You overload the suit's locking mechanism.")
```
Comment (tigercat2000) [inline code/modules/clothing/under/color.dm]: no color macros

**Comment by Krausus** [comment]
> Re: Conditions, see SOP.

I looked through SOP and space law. I couldn't find anything about parole or a similar concept. Are you sure parole is even a thing? A major reason for brig time is to punish players who misbehave, and turning brig time into a minor slap on the wrist seems contrary to that.

If it's not actually part of SOP, that would probably explain why you don't see it happen very often...

> better to be unable to change jobs for 10m than to be stuck in a brig cell for that time
> being out on parole gives you INFINITELY more freedom than being locked in a cell for the same amount of time
> So that person can have more fun in the round, rather than wasting time in a very small prison cell
> it makes for a more fun round for everyone

Getting brigged **isn't supposed to be fun.** It's a **punishment** to deter misbehavior. To assume that players, who have done something to get brigged, deserve to immediately get let back out, is missing the point of the brig entirely.

Also, who would go through the trouble of setting someone up with one of the only two parole jumpsuits to speed up their agonizing *ten minute* sentence?

> Parolees can still mine (doesn't stop you changing hardsuit)

There are machines that you must insert your ID into, after physically removing it from your jumpsuit, to use. This includes machines to earn or redeem mining points.

**Comment by Kyep** [comment]
@sasanek12 @ssirius @kasparoVy @Da-Dman234 @DaveTheHeadcrab @FlattestGuitar
Parole jumpsuits have been deleted from this PR. Only the implant changes remain.

**Comment by Faryaa** [comment]
Good idea. Thats why I proposed make either new job or extend IAA as Parole officer. Give them portable traking equipment and other things.
http://nanotrasen.se/phpBB3/viewtopic.php?f=12&t=8007

**Comment by TullyBurnalot** [comment]
I cannot overstate how much I want this to be a thing. Not only would this allow Security a bit more breathing room for your run of the mill bar brawlers, it would actually add some semblance of a deterrent for Parolees.

Plus, actually give tracking implants more use.

File: code/game/machinery/computer/prisoner.dm [hunk 4]
```diff
@@ -58,12 +57,17 @@
 			Tr = get_turf(T)
 			if((Tr) && (Tr.z != src.z))	continue//Out of range
 			if(!T.implanted) continue
-			var/loc_display = "Unknown"
 			var/mob/living/carbon/M = T.imp_in
+			var/loc_display = "Unknown"
+			var/health_display = "OK"
+			var/total_loss = round(M.getOxyLoss() + M.getToxLoss() + M.getFireLoss() + M.getBruteLoss(), 1)
```
Comment (TheDZD) [inline code/game/machinery/computer/prisoner.dm]: You could probably just do `maxhealth - health`

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -629,8 +629,10 @@ BLIND     // can't see anything
 		//makes sure that the clothing is equipped so that we can't drag it into our hand from miles away.
 		if (!(src.loc == usr))
 			return
-
 		if (!( usr.restrained() ) && !( usr.stat ) && ( over_object ))
+			if (flags & NODROP)
```
Comment (TheDZD) [inline code/modules/clothing/clothing.dm]: `if(!usr.canUnEquip(src))`

**Comment by Kyep** [comment]
@TheDZD Changed

**Comment by Kyep** [comment]
@TheDZD Changed

**Comment by TheDZD** [comment]
Conflict.

File: code/modules/clothing/clothing.dm [hunk 6]
```diff
@@ -629,8 +629,10 @@ BLIND     // can't see anything
 		//makes sure that the clothing is equipped so that we can't drag it into our hand from miles away.
 		if (!(src.loc == usr))
 			return
-
 		if (!( usr.restrained() ) && !( usr.stat ) && ( over_object ))
+			if (!usr.canUnEquip(src))
```
Comment (tigercat2000) [inline code/modules/clothing/clothing.dm]: no space between `if` and `(`

**Comment by Kyep** [comment]
@TheDZD Conflict resolved.

**Comment by Lady-Luck** [comment]
Hmm, is this going to get merged? The conflict is resolved, is there something else wrong with it?

**Comment by Kyep** [comment]
I would like it to be merged. The issues that people raised previously were addressed. The merge conflict was resolved. CL exists and is up-to-date. It also has a supermajority of support in voting.

**Comment by Krausus** [comment]
I still disagree with giving security pre-filled ether implants just because they're too lazy to get empty chem implants filled. Trying to remove inter-departmental cooperation because "uguu, working with the peasants outside security is *too much trouble*" is just bad.

You've added several pre-filled implanters, rather than the single implanter, implant pad (to configure their IDs), and multiple cased implants used by other implant boxes. Also, this increases the number of tracking implants and chem implants sec starts with; if you intend to replace the existing implants with better ones, then actually replace them.

I'm not sure about the change to the tracking implant UI. Showing the person's name seems like a poor decision, since that can change arbitrarily, though the chem implant UI already does that. Also, your change appears to add a bunch of line breaks. What does it actually look like in-game?

**Comment by Kyep** [comment]
@Krausus It isn't just laziness. Chem implants have multiple other issues. Examples: 1) Nobody knows how to use them. Even when I deliberately set out to use them, I have to spend a few minutes re-figuring out how they work every time, because it is not remotely obvious or even intuitive. 2) Prisoners fear chem implants, assuming they're lethal by default. 3) Security *could* ask for stuff from Chem, but the fact is they don't, and waiting while that happens just delays release, and makes parole less likely. It is a net loss for everyone.

As to increasing the number of implants, yes, it does, but A) This was not intended to replace, and B) given how nobody uses the existing chem implants, I did not see a reason to delete them. 

The reason the UI shows their name is that without it, you just have an ID number, and that's extremely unhelpful if you've implanted more than one person and have to make a decision regarding which one to activate. I'll take a screenshot of the UI when I get a moment.

**Comment by Krausus** [comment]
> It isn't just laziness.
> Nobody knows how to use them.

Because they're too lazy to learn.

> Prisoners fear chem implants, assuming they're lethal by default.

So? It's a brig, not a daycare. It doesn't matter if they're afraid.

> Security could ask for stuff from Chem, but the fact is they don't

And they shouldn't be rewarded for being lazy.

> and waiting while that happens just delays release, and makes parole less likely

If you're so paranoid about someone committing another crime that you have to thoroughly implant them before letting them out on parole, maybe you *shouldn't* be in a rush to get them back out the door.

**Comment by Lady-Luck** [comment]
Asking stuff from chem isn't often practical because chem tends to finish up early in the round and criminals tend to come in mid-round to late-round. That's assuming there's even a chem to begin with and it wasn't just done by MDs with the CMO's permission.

**Comment by Kyep** [comment]
Krausus, ultimately: the goal is to get more people out on Parole.
Its more fun for everyone to have at least some people serving their sentence in the community, where they can interact, potentially do useful things, etc.
Security COULD let people out on Parole now, but the fact is they never, ever do, because chem implants are massive hassle to use, and tracking implants are just terrible. All you get on the console id an ID number and a set of co-ordinates. That's not useful.
This PR adds a subtype of chem implants that might actually have a chance of seeing use, and makes tracking implants give their output in terms of human-readable words, not just cryptic numbers.
This is a game, games should be fun, and the current implementation of chem/tracking implants is awkward and unfun.
I don't see this as rewarding laziness. In fact, I see this as rewarding courage. This PR rewards security officers who want to give someone parole, accepting some loss of control in exchange for keeping the round more interesting for everyone. Implants aren't the safe option here - being stuck in a tiny cell is. Currently, the balance is so far out of whack between brigging and parole that implant-based parole basically never happens. This PR makes parole more of viable option, encouraging people to give up a little control and allow more fun in the round.

**Comment by Krausus** [comment]
> chem implants are massive hassle to use

This is a massive overdramatization. The "hassle" is that you don't want to have to interact with chemistry to make easy-mode nap-time implants.

> tracking implants are just terrible. Showing an ID number and a set of co-ordinates is not useful.

The tracking console shows an ID number and area. And you can use the teleporter to literally teleport straight on top of someone with a tracking implant.

> I don't see this as rewarding laziness. In fact, I see this as rewarding courage.

what.

> This PR rewards security officers who want to give someone parole, accepting some loss of control

*what.* These implants explicitly give you **more** control over someone than you'd have if they simply finished their sentence and walked out. You want to release someone and be able to track their location and put them to sleep at any point in the rest of the round. This drastically reduces the effort to powergame as security.

> Its more fun for everyone
> This is a game, games should be fun
> allow more fun in the round.

Alright, *stop.* The brig exists so people who commit crimes are excluded from fun for a period of time. If they don't want that to happen, they either don't commit crimes, or they don't get caught. Letting them out without any sort of punishment runs contrary to the entire point of the brig, and letting them out full of implants that trivialize tracking and recapturing antagonists is practically just powergaming.

You know what will make the round more fun? If people stopped being jerks and getting themselves brigged. You know what will help stop that? Forcing them to wait around in the brig, not having fun, until they consider not being jerks next time.

**Comment by Lady-Luck** [comment]
Parole is not just for security's sake, it's to help keep people that were arrested in the round and not afk watching a youtube. Making it easy shouldn't be a big deal.

**Comment by Kyep** [comment]
The hassle isn't just interacting with chem. In fact, in the times I've used them, interacting with chem has been the intuitive part. The hard part is getting them filled correctly. I (as medchemist) have provided ether in two minutes, only to watch Sec spend more than 5 minutes fiddling around with them trying to work out how to use the implant.

The tracking console, last I checked, did not show area, only co-ordinates. It is only with this console it shows area.

These parole implants don't exist for antags. They exist as an alternative to brig time for people who're very unlikely to commit addition violations anyway. If sec don't trust someone not to re-offend, they'll just give them the brig time instead of these.

The purpose of the brig is not to exclude people from fun. That's like saying the purpose of real-life prisons is to make people miserable. Sure, they might do that, but their actual purpose is managing and reducing crime. Similarly, the actual purpose of the brig in SS13 is to minimize space law violations. I would argue that a non-violent offender unlikely to repeat their crime, being given parole, has a lower chance of re-offending than someone who'se callously thrown in the brig for the maximum time. The latter breeds resentment and a 'screw security!' attitude. The former provides the person an incentive to be on their best behavior.

**Comment by Krausus** [comment]
> The hard part is getting them filled correctly.

Now these excuses are just getting silly. Add instructions, either to the implant case description itself, or to the box they come in. The solution to players not knowing how to do a task isn't to *remove the task.* That's insane.

> The tracking console, last I checked, did not show area, only co-ordinates

Tracking console shows area. What you're thinking about is the locator, included in the tracking implant box, which shows coordinates specifically so you can directly zero in on someone's location. You aren't modifying the locator's UI at all.

> These parole implants don't exist for antags. They exist as an alternative to brig time for people who're very unlikely to commit addition violations anyway.

You don't get to decide what they get used for - the players do. And no sane player will waste one of the three pairs of implants in their convenient new "Fuck Antagonists" box on someone who's unlikely to commit more crimes; they'll save them for the obvious antagonists that they would probably execute if it wasn't literally against the rules to do so on a whim. But now, they can simply say, "we're putting you on parole, which means we're allowed to shove these implants in you that let us ruin your day if we notice you trying to complete an antag objective. Er, I mean, commit another crime. Have fun!"

You can't just say "but these are for X!" and completely ignore the actual consequences of what something does.

> The purpose of the brig is not to exclude people from fun.
> the actual purpose of the brig in SS13 is to minimize space law violations.

Brigging is a punishment for misbehavior. It boils down to, "for what you've done, you don't get to play for X minutes". You can explain it using whatever flowery verbiage you want, but it is what it is.

> I would argue that a non-violent offender unlikely to repeat their crime, being given parole, has a lower chance of re-offending than someone who'se callously thrown in the brig for the maximum time

If someone is non-violent and unlikely to repeat their crime, you probably don't need to implant them with implants that scream, "we don't actually trust you, but thankfully, we can screw you over just in case". You just need to *let them leave.* What you say these implants are for, and what purpose they actually serve, is directly contradictory. You don't need to be implanting people you "trust". These will get used solely on people who *aren't* trusted.

Perhaps what you should be doing, instead of trying to make parole easier by buffing security through code changes, is getting parole *without implants* codified into space law. If parole is meant for people unlikely to re-offend, they don't need implants.

**Comment by Lady-Luck** [comment]
>Brigging is a punishment for misbehavior. It boils down to, "for what you've done, you don't get to play for X minutes". You can explain it using whatever flowery verbiage you want, but it is what it is.

Uhh, we want antagonists to misbehave. We just need some sort of penalty for them getting caught, otherwise it just doesn't make any sense.

A chem implant is a sufficient penalty. We want parole to be a more convenient option so it is used more.

**Comment by Krausus** [comment]
> Uhh, we want antagonists to misbehave. We just need some sort of penalty for them getting caught

We **have** a penalty. It's time in the brig, eating into the time you have left to complete your objectives. Getting a couple implants that make it trivial to track and immobilize you is actually a harsher penalty than simply losing some time.

**Comment by Lady-Luck** [comment]
>We have a penalty. It's time in the brig, eating into the time you have left to complete your objectives. Getting a couple implants that make it trivial to track and immobilize you is actually a harsher penalty than simply losing some time.

That's highly subjective, and I doubt these implants will be used frugally. This is a great option for non-violent criminals who would otherwise be perma'd.

**Comment by Krausus** [comment]
> This is a great option for non-violent criminals who would otherwise be perma'd.

Ending up in perma requires an *incredibly serious* offense - or, more likely, *multiple* serious offenses. Would you honestly label such a person as "unlikely to re-offend, suitable for parole"?

**Comment by Lady-Luck** [comment]
>Ending up in perma requires an incredibly serious offense - or, more likely, multiple serious offenses. Would you honestly label such a person as "unlikely to re-offend, suitable for parole"?

With a tracking chip in their head that can knock them out at any time? Yeah, I could completely see certain criminals being regarded save to release from perma with that.

**Comment by Krausus** [comment]
> With a tracking chip in their head that can knock them out at any time? Yeah, I could completely see certain criminals being regarded save to release from perma with that.

So what you're saying is, this one-two punch of implants is so hilariously overpowered that, so long as they were inside someone responsible for even capital-grade crimes, you'd feel comfortable letting that person walk out of the brig? Interesting.

**Comment by Lady-Luck** [comment]
Goodness gracious you like to put words in people's mouths, don't you? I think it is good to have something that requires oversight to let non-violent criminals go when they would otherwise be removed from the round.

Please stop being so hostile. We get it, you disagree. It's not the end of the world if this gets pushed.

**Comment by Fox-McCloud** [comment]
Let's keep it civil, please---all sides.

**Comment by Lady-Luck** [comment]
Well, that sucks.

**Comment by Kyep** [comment]
@Fox-McCloud @TheDZD 

Parole implants deleted.
All this PR does now is make Tracking Implants slightly more useful by showing area and health status on prisoner console. 
Description and changelog updated accordingly.

**Comment by FreeStylaLT** [comment]
From hundreds of lines to 15

the rise and fall of parole. At this point, it's become a QoL addition to tracking implants

**Comment by Lady-Luck** [comment]
It's really quite sad. :/

**Comment by Krausus** [comment]
Could you make the console format its information into a table? The old method of just throwing some vertical bars between each bit of info was lazy, but just throwing some newlines between each bit of info kinda looks worse.

Also, you mentioned that the coordinates shown on the locator are hard to use; have you considered changing it so it instead shows a distance and direction for each tracking implant?

**Comment by Kyep** [comment]
The *s render as a line in github. In game, they're the same as they are now.
I suppose I could make it into a fancy table, but that's a lot of work for a list that's not going to have many entries. I did not think it to be worth it.

I haven't changed anything about the locator. If you want to, feel free. This only changes the prisoner management console. The locator is generally not used.

**Pull request merged**